### ISSUE TITLE
Rule Request: Enforce multiple protocol inheritance style

### ISSUE THREAD

**Issue by sindresorhus** [realm/SwiftLint#3602]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

There are two ways to inherit from multiple protocols:

```swift
protocol Foo: Hashable, Identifiable {}
protocol Foo: Hashable & Identifiable {}
```

Source: https://forums.swift.org/t/is-this-a-good-idea-protocol-identifiablehashable-hashable-identifiable/47496/11

> 1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

It would be good to be able to enforce a style to make code more consistent.

> 2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

That depends on which style you choose.

Either of these would trigger:

```swift
protocol Foo: Hashable, Identifiable {}
protocol Foo: Hashable & Identifiable {}
```

These would not trigger

```swift
protocol Foo: Identifiable {}
protocol Foo: Hashable {}
```

> 3. Should the rule be configurable, if so what parameters should be configurable?

Yes, you should be able to choose which style to use.

> 4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

No
